### PR TITLE
Qualimap: BamQC: add percentage on target

### PR DESCRIPTION
**Pull request by FerriolCalvet**
Solving this issue: #2019 

- [x] This comment contains a description of changes (with reason)
This is explained in the issue referenced above. But basically I am adding an extra column to the output table of Qualimap BAMQC. This value is already present in the default output of Qualimap BAMQC. It is the **(97.14%)**.
image

- [ ] `CHANGELOG.md` has been updated

### CODE REVIEW

File: multiqc/modules/qualimap/QM_BamQC.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def parse_genome_results(self, f):
         self.general_stats_data[s_name]["mean_coverage"] = d["mean_coverage"]
         self.general_stats_data[s_name]["regions_size"] = d["regions_size"]
         self.general_stats_data[s_name]["regions_mapped_reads"] = d["regions_mapped_reads"]
+        d["regions_percentage_inside"] = (d["regions_mapped_reads"] / d["mapped_reads"]) * 100
+        self.general_stats_data[s_name]["regions_percentage_inside"] = d["regions_percentage_inside"]
```
Comment (vladsavelyev) [inline multiqc/modules/qualimap/QM_BamQC.py:145]: I'd rename it slightly for consistency with other percentage metrics
```suggestion
        d["percentage_aligned_on_target"] = (d["regions_mapped_reads"] / d["mapped_reads"]) * 100
        self.general_stats_data[s_name]["percentage_aligned_on_target"] = d["percentage_aligned_on_target"]
```

File: multiqc/modules/qualimap/QM_BamQC.py [hunk 2]
```diff
@@ -628,6 +630,14 @@ def general_stats_headers(self):
         "shared_key": "read_count",
         "hidden": True,
     }
+    self.general_stats_headers["regions_percentage_inside"] = {
```
Comment (vladsavelyev) [inline multiqc/modules/qualimap/QM_BamQC.py:633]:
```suggestion
    self.general_stats_headers["percentage_aligned_on_target"] = {
```

**Review by vladsavelyev — APPROVED** [review]
Thank you! Looks good to me. 

I think it make sense to change the title of the existing `regions_mapped_reads` metric from `"{} Aligned"` to `"{} On target"`, otherwise it has the same name as the overall number of mapped reads.

**Review by vladsavelyev — CHANGES_REQUESTED** [review]
Just rename the `regions_mapped_reads` metric from "{} Aligned" to "{} On target", please :)

**Review by vladsavelyev — APPROVED** [review]

File: multiqc/modules/qualimap/QM_BamQC.py [hunk 3]
```diff
@@ -141,6 +141,8 @@ def parse_genome_results(self, f):
         self.general_stats_data[s_name]["mean_coverage"] = d["mean_coverage"]
         self.general_stats_data[s_name]["regions_size"] = d["regions_size"]
         self.general_stats_data[s_name]["regions_mapped_reads"] = d["regions_mapped_reads"]
+        d["percentage_aligned_on_target"] = (d["regions_mapped_reads"] / d["mapped_reads"]) * 100
```
Comment (ewels) [inline multiqc/modules/qualimap/QM_BamQC.py:144]: It's good to force a float by doing `* 100.0`

File: multiqc/modules/qualimap/QM_BamQC.py [hunk 4]
```diff
@@ -141,6 +141,8 @@ def parse_genome_results(self, f):
...
+        self.general_stats_data[s_name]["percentage_aligned_on_target"] = d["percentage_aligned_on_target"]
```
Comment (ewels) [inline multiqc/modules/qualimap/QM_BamQC.py:145]: This is ripe for a `ZeroDivisionError`!

Also, I'm not sure that it helps much to break this out into a new key in `d`?

File: multiqc/modules/qualimap/QM_BamQC.py [hunk 5]
```diff
@@ -622,12 +624,20 @@ def general_stats_headers(self):
...
     }
     self.general_stats_headers["regions_mapped_reads"] = {
-        "title": "{} Aligned".format(config.read_count_prefix),
+        "title": "{} On target".format(config.read_count_prefix),
         "description": "Number of mapped reads on target region ({})".format(config.read_count_desc),
         "scale": "RdYlGn",
         "shared_key": "read_count",
         "hidden": True,
     }
+    self.general_stats_headers["percentage_aligned_on_target"] = {
```
Comment (ewels) [inline multiqc/modules/qualimap/QM_BamQC.py:633]: I think it makes sense to have this to the left of the more generic columns like "M Aligned", as it's a subset. So better to move further up the code basically.

**Review by ewels — APPROVED** [review]
Looks good! Couple of minor pedantic comments, will fix quickly myself now.

**Comment by ewels** [comment]
I realised that several of my review comments also applied to existing code within the module! So was a good opportunity to clean those bits up as well ✅ 

Many thanks for this @FerriolCalvet!

**Pull request merged**